src/parser/tptp/tptp.cpp

```diff
@@ -213,14 +213,21 @@ void Tptp::checkLetBinding(std::vector<Expr>& bvlist, Expr lhs, Expr rhs, bool f
   std::sort(v.begin(), v.end());
   std::sort(bvlist.begin(), bvlist.end());
   // ensure all let-bound variables appear on the LHS, and appear only once
-  for(size_t i = 0; i < bvlist.size(); ++i) {
-    std::vector<CVC4::Expr>::const_iterator found = std::lower_bound(v.begin(), v.end(), bvlist[i]);
-    if(found == v.end() || *found != bvlist[i]) {
-      parseError("malformed let: LHS must make use of all quantified variables, missing `" + bvlist[i].toString() + "'");
+  for (size_t i = 0; i < bvlist.size(); ++i) {
+    std::vector<CVC4::Expr>::const_iterator found =
+        std::lower_bound(v.begin(), v.end(), bvlist[i]);
+    if (found == v.end() || *found != bvlist[i]) {
+      parseError(
+          "malformed let: LHS must make use of all quantified variables, "
+          "missing `" +
+          bvlist[i].toString() + "'");
     }
-    std::vector<CVC4::Expr>::const_iterator found2 = found + 1;
-    if(found2 != v.end() && *found2 == *found) {
-      parseError("malformed let: LHS cannot use same bound variable twice: " + (*found).toString());
+    if (found != v.end()) {
```

Comment:
If `found == v.end()`, then we'd output the first parse error and quit no? So `found` should never be equal to `v.end()` here.

Reply:
See the coverity scan report for CID 1172286. It is not able to infer that parseError's contract is that it never returns.

TBH I don't see the harm in having the extra sanity check here.

Reply:
How about an assert instead of an `if` statement? That would seem clearer to me.

Reply:
Went with the suggestion. This is not great as the check probably should not be compiled out, but should solve the problem.

A nicer solution might be to avoid this by marking parseError as [[noreturn]]. but this seems to be unsupported by SWIG, http://www.swig.org/Doc3.0/CPlusPlus11.html#CPlusPlus11_attributes .

Reply:
Yeah, it is not great that the check gets compiled out for production builds but at least you'll be able to catch it with a debug build (and short of a compiler bug, this assertion should not fail). The original patch would have silently continued (even in a debug build), so this is slightly better IMHO :)